File: HackerRank/Encryption.c
https://www.hackerrank.com/challenges/encryption/submissions/code/216927428

#include<stdio.h>
#include<math.h>
int main()
{
        char s[82];
        scanf("%s",s);
        int i,c,r,n,j,k,sq;
        for(i=0;s[i];++i)
         sq=sqrt(i);
         if(sq*sq==i)
         {
                c=r=sq;
         }
         else
         {       
                r=sq;
                c=sq+1;
                if(r*c<i)
                {
                    r=sq+1;c=sq+1;
                }
         }
        

        for(j=0;j<c;++j)
        {       
         for(k=0;k<r;++k)
         {
            if(j+k*c<i)
                  printf("%c",s[j+k*c]);
                }
         printf(" ");
        }  

}

File: HackerRank/Encryption.go
package main

import (
	"fmt"
	"math"
	"strings"
)

// link to problem: https://www.hackerrank.com/challenges/encryption/problem

func encryption(s string) string {
	s = strings.ReplaceAll(s, " ", "")
	lenStr := float64(len(s))
	min := math.Floor(math.Sqrt(lenStr))
	max := math.Ceil(math.Sqrt(lenStr))
	if min*max < lenStr {
		min++
	}

	var temp [][]string
	var rows []string
	for i, v := range s {
		if i%int(max) == 0 && i >= int(max) {
			temp = append(temp, rows)
			rows = []string{}
		}
		rows = append(rows, string(v))
	}
	temp = append(temp, rows)

	var result string
	for x := 0; x < int(max); x++ {
		for y := 0; y < int(min); y++ {
			if len(temp[y]) < int(max) && x >= len(temp[y]) {
				break
			}
			result += temp[y][x]
		}
		result += " "
	}
	return result
}

func main() {
	fmt.Println(encryption("haveaniceday"))
	fmt.Println(encryption("feedthedog"))
	fmt.Println(encryption("chillout"))
}
